### PR TITLE
[RateLimiter] Moved classes implementing LimiterInterface to a new namespace

### PR DESCRIPTION
**Pull request by Nyholm**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no?
| Tickets       | 
| License       | MIT
| Doc PR        | 

Before we release the RateLimit component. 
I think it would be a good idea to put the 7 classes that belongs to a specific strategy in their own "Policy" namespace. It is very likely that it will be more strategies in the future and the `Symfony\Component\RateLimiter` namespace is crowed as it is. 

I decided not to put the `CompoundLimiter` in this namespace as it is not a strategy.

### CODE REVIEW

**Review by stof — APPROVED** [review]

File: src/Symfony/Component/RateLimiter/Tests/Strategy/TokenBucketLimiterTest.php [hunk 1]
```diff
@@ -9,16 +9,16 @@
  * file that was distributed with this source code.
  */
 
-namespace Symfony\Component\RateLimiter\Tests;
+namespace Symfony\Component\RateLimiter\Strategy\Tests;
```
Comment (stof) [inline src/Symfony/Component/RateLimiter/Tests/Strategy/TokenBucketLimiterTest.php]: wrong change. Should be `Tests\Strategy`

Comment (Nyholm) [reply src/Symfony/Component/RateLimiter/Tests/Strategy/TokenBucketLimiterTest.php]: Thank you

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -31,7 +31,7 @@
 use Symfony\Component\Messenger\MessageBusInterface;
 use Symfony\Component\Notifier\Notifier;
 use Symfony\Component\PropertyInfo\PropertyInfoExtractorInterface;
-use Symfony\Component\RateLimiter\TokenBucketLimiter;
+use Symfony\Component\RateLimiter\Strategy\TokenBucketLimiter;
```
Comment (jderusse) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: Given the interface is named `LimiterInterface`, shouldn't it be `Symfony\Component\RateLimiter\Limiter` ?

Currently all namespace named `Strategy` are related to a XxxStrategyInterface. (`Lock/StrategyInterface`, `Workflow/SupportStrategyInterface`, `HttpClient/RetryStrategyInterface`, `Asset/VersionStrategyInterface`, `Messenger/RetryStrategyInterface`

Comment (Nyholm) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: Maybe.. The `Symfony\Component\RateLimiter` namespace has so much "limit" or "rate limit" related names already.

Im not super passionate about any name. I just wanted to move these classes to a separate namespace.

Comment (javiereguiluz) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: I can understand both arguments ... but I'd prefer `Strategy/`.

**Review by javiereguiluz — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php (hunk 2 above)
Comment (wouterj) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: :+1: for strategy as well.

A (last?) change to this component would involve renaming all "strategy" to "policy" to follow common rate limit terminology (or at least, that of the rate limit HTTP RFC).

Comment (fabpot) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: I agree with Wouter that having names more similar to the HTTP RFC would be a good thing.

Comment (Nyholm) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: Done. I renamed namespace and config to "policy"

**Comment by nicolas-grekas** [comment]
Failure is related, isn't it?

**Comment by Nyholm** [comment]
Yes, they are. Im working on it

**Comment by Nyholm** [comment]
The tests looks much happyr now. 

It is ready for merge.

File: src/Symfony/Bundle/SecurityBundle/Tests/Functional/FormLoginTest.php [hunk 3]
```diff
@@ -123,6 +123,7 @@ public function testLoginThrottling($username, $password)
         $form['_username'] = $username;
         $form['_password'] = $password;
         $client->submit($form);
+        $x = $client->getResponse();
```
Comment (wouterj) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/FormLoginTest.php]: Can be removed

File: src/Symfony/Component/RateLimiter/Policy/TokenBucketLimiter.php [hunk 4]
```diff
@@ -9,11 +9,16 @@
...
  */
 
-namespace Symfony\Component\RateLimiter;
+namespace Symfony\Component\RateLimiter\Policy;
 
 use Symfony\Component\Lock\LockInterface;
 use Symfony\Component\Lock\NoLock;
 use Symfony\Component\RateLimiter\Exception\MaxWaitDurationExceededException;
+use Symfony\Component\RateLimiter\LimiterInterface;
+use Symfony\Component\RateLimiter\Rate;
```
Comment (wouterj) [inline src/Symfony/Component/RateLimiter/Policy/TokenBucketLimiter.php]: I think `Rate` should maybe also be moved in the new `Policy` namespace? It's only used when creating a `TokenBucketLimiter`.

File: src/Symfony/Component/RateLimiter/Policy/TokenBucketLimiter.php [hunk 5]
```diff
@@ -9,11 +9,16 @@
...
+use Symfony\Component\RateLimiter\RateLimit;
+use Symfony\Component\RateLimiter\Reservation;
+use Symfony\Component\RateLimiter\ResetLimiterTrait;
```
Comment (wouterj) [inline src/Symfony/Component/RateLimiter/Policy/TokenBucketLimiter.php]: The `ResetLimiterTrait` should also be moved - it's used only by the policy classes.

**Review by wouterj — APPROVED** [review]
Thank you Tobias! Some final remarks, but I think these can also be fixed while merging.

File: src/Symfony/Bundle/SecurityBundle/Tests/Functional/FormLoginTest.php (hunk 3 above)
Comment (Nyholm) [reply src/Symfony/Bundle/SecurityBundle/Tests/Functional/FormLoginTest.php]: Oops =)

File: src/Symfony/Component/RateLimiter/Policy/TokenBucketLimiter.php (hunk 5 above)
Comment (Nyholm) [reply src/Symfony/Component/RateLimiter/Policy/TokenBucketLimiter.php]: Sure. Will do

File: src/Symfony/Component/RateLimiter/Policy/TokenBucketLimiter.php (hunk 4 above)
Comment (Nyholm) [reply src/Symfony/Component/RateLimiter/Policy/TokenBucketLimiter.php]: I agree.

**Comment by fabpot** [comment]
Thank you @Nyholm.

**Comment by Nyholm** [comment]
Thank you for the reviews and merge!

**Pull request merged**